### ISSUE TITLE
frida-tools 12.4.2 wheel build fails

### ISSUE THREAD

**Issue by seanthegeek** [frida/frida-tools#158]
OS: REMnux 2024.21.5 (Based on Ubuntu 20.04)
Python: 3.8

```
sudo pip3 install -U frida-tools
Requirement already satisfied: frida-tools in /usr/local/lib/python3.8/dist-packages (12.3.0)
Collecting frida-tools
  Using cached frida-tools-12.4.2.tar.gz (345 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "/usr/local/lib/python3.8/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/usr/local/lib/python3.8/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/usr/local/lib/python3.8/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-jtgnfn44/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-jtgnfn44/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-jtgnfn44/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 487, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-jtgnfn44/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 99, in <module>
      TypeError: 'type' object is not subscriptable
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by digitalsleuth** [comment]
For additional detail, this actually occurs in all 12.4.x installs. The last version which does not have this error is 12.3.0.
```bash
root@922613b4a07b:/# python3 -m pip install -U frida-tools==12.4.0
Collecting frida-tools==12.4.0
  Downloading frida-tools-12.4.0.tar.gz (55 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.1/55.1 kB 1.1 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "/usr/local/lib/python3.8/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/usr/local/lib/python3.8/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/usr/local/lib/python3.8/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-xz5k4wub/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-xz5k4wub/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-xz5k4wub/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 487, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-xz5k4wub/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 97, in <module>
      TypeError: 'type' object is not subscriptable
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
root@922613b4a07b:/# python3 -m pip install -U frida-tools==12.3.0
Collecting frida-tools==12.3.0
  Downloading frida-tools-12.3.0.tar.gz (200 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 200.5/200.5 kB 2.6 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting colorama<1.0.0,>=0.2.7 (from frida-tools==12.3.0)
  Downloading colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Collecting frida<17.0.0,>=16.0.9 (from frida-tools==12.3.0)
  Downloading frida-16.2.5-cp37-abi3-manylinux_2_5_x86_64.whl.metadata (2.0 kB)
Collecting prompt-toolkit<4.0.0,>=2.0.0 (from frida-tools==12.3.0)
  Downloading prompt_toolkit-3.0.43-py3-none-any.whl.metadata (6.5 kB)
Collecting pygments<3.0.0,>=2.0.2 (from frida-tools==12.3.0)
  Downloading pygments-2.18.0-py3-none-any.whl.metadata (2.5 kB)
Collecting typing-extensions (from frida<17.0.0,>=16.0.9->frida-tools==12.3.0)
  Downloading typing_extensions-4.11.0-py3-none-any.whl.metadata (3.0 kB)
Collecting wcwidth (from prompt-toolkit<4.0.0,>=2.0.0->frida-tools==12.3.0)
  Downloading wcwidth-0.2.13-py2.py3-none-any.whl.metadata (14 kB)
Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Downloading frida-16.2.5-cp37-abi3-manylinux_2_5_x86_64.whl (29.3 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 29.3/29.3 MB 68.6 MB/s eta 0:00:00
Downloading prompt_toolkit-3.0.43-py3-none-any.whl (386 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 386.1/386.1 kB 25.5 MB/s eta 0:00:00
Downloading pygments-2.18.0-py3-none-any.whl (1.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 42.0 MB/s eta 0:00:00
Downloading typing_extensions-4.11.0-py3-none-any.whl (34 kB)
Downloading wcwidth-0.2.13-py2.py3-none-any.whl (34 kB)
Building wheels for collected packages: frida-tools
  Building wheel for frida-tools (pyproject.toml) ... done
  Created wheel for frida-tools: filename=frida_tools-12.3.0-py3-none-any.whl size=209494 sha256=18ca033ba77aa038ceed60bd55d7296614e3dc9c85ba824ad731f0b95c54e359
  Stored in directory: /root/.cache/pip/wheels/a3/55/ba/d47fc0292871c146f5522f655a9a98698fa33ca85caedfdb08
Successfully built frida-tools
Installing collected packages: wcwidth, typing-extensions, pygments, prompt-toolkit, colorama, frida, frida-tools
Successfully installed colorama-0.4.6 frida-16.2.5 frida-tools-12.3.0 prompt-toolkit-3.0.43 pygments-2.18.0 typing-extensions-4.11.0 wcwidth-0.2.13
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
root@922613b4a07b:/# python3 -m pip list | egrep 'setuptools|wheel|pip'
pip                24.0
setuptools         70.0.0
wheel              0.43.0
```

**Comment by UndefinedIdentifier** [comment]
you could try this fix
https://github.com/frida/frida-tools/pull/159

**Comment by oleavr** [comment]
Oops. Thanks to @UndefinedIdentifier this should now hopefully be fixed in 12.4.3, just released.
